src/Bicep.LangServer/Handlers/InsertResourceHandler.cs

```diff
@@ -79,8 +86,18 @@ public Task<Unit> Handle(InsertResourceParams request, CancellationToken cancell
                         BicepTelemetryEvent.InsertResourceFailure("ParseResourceIdFailed"),
                         Unit.Value);
                 }
-
-                var matchedType = azResourceTypeLoader.GetAvailableTypes()
+                var azProviderDeclaration = model.SourceFile.ProgramSyntax.Children
+                    .OfType<ImportDeclarationSyntax>()
+                    .FirstOrDefault(x => x.Specification.Name.Equals(AzNamespaceType.BuiltInName, LanguageConstants.IdentifierComparison));
+                var resourceTypeLoader = azResourceTypeLoaderFactory.GetResourceTypeLoader(azProviderDeclaration?.Specification.Version, model.Features);
```

Comment:
Is there a way to do this with the semantic model rather than the syntax tree?

Reply:
After following up offline I learned that this handler actually has access to the resource providers so I am using a pattern similar to the reference in `BicepCodeActionHandler`